### PR TITLE
Return the OMPS-pushed version when IIB backports a bundle

### PR DESCRIPTION
**Pull request by lipoja**
- log response from OMPS API for every successful request
- collecting operator version for all packages
- updated patch_request function to process request with omps_operator_version
- adding propagating of OMPS response to _get_base_dir_and_pkg_name
- adding omps_operator_version to RequestAdd database table model
- added set_omps_operator_version; set the set_omps_operator_version of the request using the IIB API.
- added call of set_omps_operator_version after collecting all versions

For more information refer to [CLOUDDST-2201]

### CODE REVIEW

**Comment by lipoja** [comment]
Hello @yashvardhannanavati @mprahl @lcarva @querti @midnightercz @chandwanitulsi can I ask you for code review?

File: iib/workers/tasks/legacy.py [hunk 1]
```diff
@@ -141,6 +146,9 @@ def _push_package_manifest(package_dir, cnr_token, organization):
             raise IIBError(
                 f'Push to {organization} in the legacy app registry was unsucessful: {error_msg}'
             )
+        else:
```
Comment (yashvardhannanavati) [inline iib/workers/tasks/legacy.py:149]: I don't think you need this `else`.

File: tests/test_workers/test_tasks/test_legacy.py [hunk 2]
```diff
@@ -127,17 +127,19 @@ def test_opm_index_export(mock_run_cmd):
...
+def test_export_legacy_packages(mock_oie, mock_srs, mock_soov, mock_ppm, mock_zp, mock_vpi):
+    packages = {'prometheus'}
     legacy.export_legacy_packages(packages, 3, 'from:index', 'token', 'org')
 
     mock_oie.assert_called_once()
     mock_vpi.assert_called_once()
     mock_zp.assert_called_once()
     mock_ppm.assert_called_once()
     mock_srs.assert_called_once()
+    mock_soov.assert_called_once()
```
Comment (yashvardhannanavati) [inline tests/test_workers/test_tasks/test_legacy.py:142]: optional: you could probably use `assert_called_once_with` here.

File: iib/web/models.py [hunk 3]
```diff
@@ -837,7 +839,12 @@ def from_json(cls, kwargs, batch=None):
 
         cls._from_json(
             request_kwargs,
-            additional_optional_params=['from_index', 'organization', 'bundles'],
+            additional_optional_params=[
+                'from_index',
+                'organization',
+                'bundles',
+                'omps_operator_version',
```
Comment (yashvardhannanavati) [inline iib/web/models.py:846]: Why are we allowing the user to specify `omps_operator_version` when we don't use it while calling OMPS? Am I missing something?

**Review by yashvardhannanavati — CHANGES_REQUESTED** [review]

**Review by querti — COMMENTED** [review]
Was is agreed upon how CVP will consume this data? Their merge request mentions reading state history, but this PR expects it to read the logs or querying for task metadata.

**Comment by lipoja** [comment]
> Was is agreed upon how CVP will consume this data? Their merge request mentions reading state history, but this PR expects it to read the logs or querying for task metadata.

Yes I talked with Krunoslav from CVP and he is fine this implementation.

**Review by querti — APPROVED** [review]

**Review by midnightercz — APPROVED** [review]

File: iib/web/models.py [hunk 4]
```diff
@@ -860,6 +862,10 @@ def to_json(self, verbose=True):
         rv = super().to_json(verbose=verbose)
         rv.update(self.get_common_index_image_json())
         rv['organization'] = self.organization
+        if self.omps_operator_version:
```
Comment (yashvardhannanavati) [inline iib/web/models.py:865]: You can initialize  `rv['omps_operator_version'] = {}` and then check the `if`. That way, you can remove the else block, no?

**Comment by yashvardhannanavati** [comment]
Just one comment. This is looking good. Let's merge it once it's fixed. Thanks Jan!

File: iib/web/models.py (hunk 4 above)
Comment (lipoja) [reply iib/web/models.py:865]: yes, however I am used to do it this way, because it will save you some processing time (allocating memory and creating empty dictionary) in case the `if` is true.
```python
rv['omps_operator_version'] = {}  # <--- This is done every time
if self.omps_operator_version:
    rv['omps_operator_version'] = json.loads(self.omps_operator_version)
```
vs.

```python
if self.omps_operator_version:
    rv['omps_operator_version'] = json.loads(self.omps_operator_version)
else:
    rv['omps_operator_version'] = {}  # <--- This is done only in case omps_operator_version is False
```

Comment (yashvardhannanavati) [reply iib/web/models.py:865]: Yup, but it's a battle between readability and negligible processing time, in which according to me readability wins. But I'll leave the final call to you

File: iib/workers/tasks/legacy.py [hunk 5]
```diff
@@ -120,7 +123,9 @@ def _push_package_manifest(package_dir, cnr_token, organization):
         app registry via OMPS.
     :param str organization: the organization name in the legacy app registry to which
          the backported packages should be pushed to.
-    :raises IIBError: if the push is unsucessful
+    :return: Response from OMPS in JSON format
+    :rtype: dict|None
+    :raises IIBError: if the push is unsuccessful
```
Comment (yashvardhannanavati) [inline iib/workers/tasks/legacy.py:128]: Optional: `if the push is unsuccessful` => `if the push fails`

File: tests/test_web/test_api_v1.py [hunk 6]
```diff
@@ -504,6 +505,7 @@ def test_add_bundle_success(
             'url': 'http://localhost/api/v1/builds/1/logs',
             'expiration': '2020-02-15T17:03:00Z',
         },
+        'omps_operator_version': {},
```
Comment (yashvardhannanavati) [inline tests/test_web/test_api_v1.py:508]: May I ask why is the org not None but still the `omps_operator_version` missing? Could you please add the test where both of them are not None?

File: iib/workers/tasks/legacy.py (hunk 5 above)
Comment (lipoja) [reply iib/workers/tasks/legacy.py:128]: I guess your suggestion is better

File: tests/test_web/test_api_v1.py (hunk 6 above)
Comment (lipoja) [reply tests/test_web/test_api_v1.py:508]: I guess here is tested just the first response (and not waiting for finals state). `omps_operator_version` is filled after OMPS is API called and I guess this is not the test for it.

The test for this is covered in our integration test (iib-api-tests).

Comment (yashvardhannanavati) [reply tests/test_web/test_api_v1.py:508]: You can probably mock the response of the OMPS call and return a mock value, no?

**Review by yashvardhannanavati — APPROVED** [review]

**Comment by lipoja** [comment]
re-based and squashed all commits requested by @yashvardhannanavati
